src/Ion.ts

```diff
@@ -57,7 +58,7 @@ function makeBinaryReader(span: Span, options: Options) : BinaryReader {
 }
 
 function makeTextReader(span: Span, options: Options) : TextReader {
-  return new TextReader(span, options && options.catalog);
+  return new TextReader(span, options && options.catalog, options && options.raw_tokens);
```

Comment:
I  agree with this basic approach, but I'm not prepared to deal with this right now;  I think there will be some issues splitting it based on the way the existing parsing automaton works.